jf_agent/agent_logging.py

```diff
@@ -72,3 +123,15 @@ def log_and_print(logger, level, msg, exc_info=False):
     print(msg, flush=True)
     if exc_info:
         print(traceback.format_exc())
+
+
+def log_and_print_error_or_warning(logger, level, error_code,msg_args=[], exc_info=False):
```

Comment:
Looks like some formatting inconsistencies here -- no space after the comma -- you might not have black configured in your editor or virtualenv... you can do that or run it command-line with `black .`.